src/examples/USEnrichmentSecondaryExample.cs

```diff
@@ -0,0 +1,116 @@
+namespace Examples
+{
+	using System;
+	using System.Net;
+	using System.Collections.Generic;
+	using System.IO;
+    using System.Linq;
+    using SmartyStreets;
+	using SmartyStreets.USEnrichmentApi;
+    using System.Reflection;
+    using System.Text;
+
+	internal static class USEnrichmentSecondaryExample
+	{
+		public static void Run()
+		{
+            // specifies the TLS protocoll to use - this is TLS 1.2
+            const SecurityProtocolType tlsProtocol1_2 = (SecurityProtocolType)3072;
+
+            // var authId = "Your SmartyStreets Auth ID here";
+            // var authToken = "Your SmartyStreets Auth Token here";
+
+            // We recommend storing your keys in environment variables instead---it's safer!
+            var authId = Environment.GetEnvironmentVariable("SMARTY_AUTH_ID");
+			var authToken = Environment.GetEnvironmentVariable("SMARTY_AUTH_TOKEN");
+			ServicePointManager.SecurityProtocol = tlsProtocol1_2;
+
+			var client = new ClientBuilder(authId, authToken).BuildUsEnrichmentApiClient();
+			
+            // See the US Enrichment API documenation for all available datasets and data subsets https://www.smarty.com/docs/cloud/us-address-enrichment-api#data-sets
+            SmartyStreets.USEnrichmentApi.Secondary.Result[] results = null;
+            var lookup = new SmartyStreets.USEnrichmentApi.Secondary.Lookup("325023201");
+            // Options available for Lookup
+            // lookup.SetEtag("HAYDKMJXHA4DKNA");
+
+            try {
+                results = client.SendSecondaryLookup(lookup);
+            }
+            catch (NotModifiedException ex) {
+                Console.WriteLine(ex.Message); // The Etag value provided represents the latest version of the requested record
+            }
+            catch (Exception ex) {
+                Console.WriteLine(ex.Message + ex.StackTrace);
+            }
+            
+            if (results != null) {
+                foreach (SmartyStreets.USEnrichmentApi.Secondary.Result result in results) {
```

Comment:
Unless you are specifically trying to demonstrate the type, you can use "var result in results" for simplicity and readability.

Reply:
Is this preferable over results is not null? The former seems more readable to me, but perhaps that's just my bias

Reply:
These changes are only compatible with later versions of C#, which may cause some of our clients to break when running the examples, which isn't ideal. It's only the example, so it wouldn't break clients existing implementations, but it may make it more difficult for newer users to run the examples easily

Reply:
Let's consider some of these changes in the future as we redo some of this SDK. For now, this is OK.